include/dirent.h

```diff
@@ -86,6 +86,14 @@
 #define DT_LNK                    DTYPE_LINK
 #define DT_SOCK                   DTYPE_SOCK
 
+#define dirent64                  dirent
+#define readdir64                 readdir
+#define readdir64_r               readdir_r
+#define scandir64                 scandir
+#define alphasort64               alphasort
+#define versionsort64             versionsort
+#define getdents64                getdents
+
```

Comment:
Please point to where this is defined in pubs.opengroup.org

If it is shims for linux maybe we should consider a different approach. 

Should this all be behind the Kconfig to get an error if config is wrong?

Reply:
getdents is Linux specific:
https://man7.org/linux/man-pages/man2/getdents.2.html
but @Donny9 is preparing a new patchset to convert dir API to the most conformable implementation in this week like musl:
https://git.musl-libc.org/cgit/musl/tree/src/dirent
The change will:

1. Convert fs/dirent to a char driver just like eventfd.
2. All dirxxx move to libs/libc/dirent

getdents will add in that patchset. The macro here is to avoid him forget to add the definition.